File: cpp/BaseComponent.cpp
#pragma once
#include <iostream>
#include <string>

using namespace std;

class BaseComponent {
protected:
    string id;
    int warrantyMonths;
public:
    BaseComponent() : id(""), warrantyMonths(0) {}
    
    BaseComponent(string id, int warrantyMonths) {
        this->id = id;
        this->warrantyMonths = warrantyMonths;
    }
    
    void setId(string id) {
        this->id = id;
    }
    
    void setWarrantyMonths(int warrantyMonths) {
        this->warrantyMonths = warrantyMonths;
    }
    
    string getId() const {
        return id;
    }
    
    int getWarrantyMonths() const {
        return warrantyMonths;
    }
    
    virtual void displayInfo() const {
        cout << "ID: " << id << ", Warranty: " << warrantyMonths << " months" << endl;
    }
    
};

File: java/BaseComponent.java
class BaseComponent {
    private String id;
    private int warrantyMonths;

    public BaseComponent() {
        this.id = "";
        this.warrantyMonths = 0;
    }

    public BaseComponent(String id, int warrantyMonths) {
        this.id = id;
        this.warrantyMonths = warrantyMonths;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setWarrantyMonths(int warrantyMonths) {
        this.warrantyMonths = warrantyMonths;
    }

    public String getId() {
        return id;
    }

    public int getWarrantyMonths() {
        return warrantyMonths;
    }

    public void displayInfo() {
        System.out.println("ID: " + id + ", Warranty: " + warrantyMonths + " months");
    }
}
